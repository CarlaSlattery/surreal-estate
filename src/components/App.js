import React from "react";

import { Routes, Route } from "react-router-dom";
import Navbar from "./Navbar";
import Properties from "./Properties";
import AddProperty from "./AddProperty";

const App = () => {
  return (
    <div className="App">
      <Navbar />
      <Routes>
        <Route path="/" element={Properties} />
        <Route path="/add" element={AddProperty} />
      </Routes>
      <AddProperty />
    </div>
  );
};

export default App;
