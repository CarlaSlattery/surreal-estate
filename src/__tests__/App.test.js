import React from "react";
import { render, screen } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";
import App from "../components/App";

describe("App", () => {
  it("renders app component correctly", () => {
    const { asFragment } = render(
      <BrowserRouter>
        <App />
      </BrowserRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
