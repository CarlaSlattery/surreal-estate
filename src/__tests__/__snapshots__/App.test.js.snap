// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders app component correctly 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <div
      class="navBar"
    >
      <img
        alt="surreal estate logo"
        class="logo"
        src="/logo.png"
      />
      <span
        class="navbar-heading"
      >
        Surreal Estate
      </span>
      <ul
        class="navbar-links"
      >
        <li
          class="navbar-links-item"
        >
          <a
            href="/"
          >
            View Properties
          </a>
        </li>
        <li
          class="navbar-links-item"
        >
          <a
            href="/add"
          >
            Add Properties
          </a>
        </li>
      </ul>
    </div>
    <div
      class="properties-container"
    >
      <div
        class="sidebar"
      >
        <div
          class="sidebar"
        >
          <a
            class="location-link"
            href="/?query={\\"city\\":\\"Leeds\\"}"
          >
            Leeds
          </a>
          <a
            class="location-link"
            href="/?query={\\"city\\":\\"Liverpool\\"}"
          >
            Liverpool
          </a>
          <a
            class="location-link"
            href="/?query={\\"city\\":\\"Manchester\\"}"
          >
            Manchester
          </a>
          <a
            class="location-link"
            href="/?query={\\"city\\":\\"Sheffield\\"}"
          >
            Sheffield
          </a>
          <div
            class="sorting-links"
          >
            <a
              class="price-link"
              href="/?sort={\\"price\\":1}"
            >
              Price Ascending
            </a>
            <a
              class="price-link"
              href="/?sort={\\"price\\":-1}"
            >
              Price Descending
            </a>
          </div>
          <div
            class="search"
          >
            <form>
              <input
                type="text"
                value=""
              />
              <button
                type="submit"
              >
                Search
              </button>
            </form>
          </div>
        </div>
      </div>
      <div
        class="property-cards"
      />
    </div>
    <div
      class="properties-container"
    >
      <div
        class="sidebar"
      >
        <div
          class="sidebar"
        >
          <a
            class="location-link"
            href="/?query={\\"city\\":\\"Leeds\\"}"
          >
            Leeds
          </a>
          <a
            class="location-link"
            href="/?query={\\"city\\":\\"Liverpool\\"}"
          >
            Liverpool
          </a>
          <a
            class="location-link"
            href="/?query={\\"city\\":\\"Manchester\\"}"
          >
            Manchester
          </a>
          <a
            class="location-link"
            href="/?query={\\"city\\":\\"Sheffield\\"}"
          >
            Sheffield
          </a>
          <div
            class="sorting-links"
          >
            <a
              class="price-link"
              href="/?sort={\\"price\\":1}"
            >
              Price Ascending
            </a>
            <a
              class="price-link"
              href="/?sort={\\"price\\":-1}"
            >
              Price Descending
            </a>
          </div>
          <div
            class="search"
          >
            <form>
              <input
                type="text"
                value=""
              />
              <button
                type="submit"
              >
                Search
              </button>
            </form>
          </div>
        </div>
      </div>
      <div
        class="property-cards"
      />
    </div>
    <div
      class="add-property"
    >
      <h2>
        Add Property
      </h2>
      <form>
        <label
          for="title"
        >
          Description:
          <input
            id="title"
            name="title"
            placeholder="description"
            value=""
          />
        </label>
        <label
          for="city"
        >
          Location:
          <select
            id="city"
            name="city"
          >
            <option
              value="Manchester"
            >
              Manchester
            </option>
            <option
              value="Liverpool"
            >
              Liverpool
            </option>
            <option
              value="Leeds"
            >
              Leeds
            </option>
            <option
              value="Sheffield"
            >
              Sheffield
            </option>
          </select>
        </label>
        <label
          for="type"
        >
          Type of property:
          <select
            id="type"
            name="type"
          >
            <option
              value="Flat"
            >
              Flat
            </option>
            <option
              value="Detached"
            >
              Detached
            </option>
            <option
              value="Semi-Detached"
            >
              Semi-Detached
            </option>
            <option
              value="Terraced"
            >
              Terraced
            </option>
            <option
              value="End of Terrace"
            >
              End of Terrace
            </option>
            <option
              value="Cottage"
            >
              Cottage
            </option>
            <option
              value="Bungalow"
            >
              Bungalow
            </option>
          </select>
        </label>
        <label
          for="bedrooms"
        >
          Bedrooms:
          <input
            id="bedrooms"
            name="bedrooms"
            placeholder="bedrooms"
            value="0"
          />
        </label>
        <label
          for="bathrooms"
        >
          Bathrooms:
          <input
            id="bathrooms"
            name="bathrooms"
            placeholder="bathrooms"
            value="0"
          />
        </label>
        <label
          for="price"
        >
          Price:
          <input
            id="price"
            name="price"
            placeholder="price"
            value="0"
          />
        </label>
        <label
          for="email"
        >
          Email:
          <input
            id="email"
            name="email"
            placeholder="email"
            type="email"
            value=""
          />
        </label>
        <button
          class="add-button"
          type="submit"
        >
          Add
        </button>
      </form>
    </div>
  </div>
</DocumentFragment>
`;
